### ISSUE TITLE
How to release disk space and memory caused by jupyter notebook?

### ISSUE THREAD

**Issue by dot27** [ipython/ipython#9436]
I am working on an linux machine (ubuntu 14.04). I am using jupyter notebook. I am working on an EC2 instacne. When trying to remotely connect to the notebook, i get the following:

OperationalError: database or disk is full
which happened after i run some notebook, which stopped due to running disk space. I want to release that disk space, caused by past notebook.

 Filesystem            Size  Used Avail Use% Mounted on
 /dev/xvda1            7.9G  7.8G   97M  99% /
How can i release the memory?

**Comment by takluyver** [comment]
You would probably need to ssh in and remove some files.

**Comment by dot27** [comment]
H,
thanks.
How can i konw what are the files to be removed?

**Comment by dot27** [comment]
The Ipython code does not create files for clarification.
This makes it harder to know which files should be deleted.

**Comment by takluyver** [comment]
We don't create many files by default - there's a history database, but it's very hard to make that very large (for me it's 2.4MB on a computer I used for two years). The notebooks themselves shouldn't be that big unless you have thousands of them.

Can you work out which files are taking up the space?

**Comment by dot27** [comment]
df -h gives me this:
Filesystem      Size  Used Avail Use% Mounted on
udev            3.9G     0  3.9G   0% /dev
tmpfs           787M  9.6M  778M   2% /run
/dev/sda8       137G   20G  110G  15% /
tmpfs           3.9G  463M  3.4G  12% /dev/shm
tmpfs           5.0M  4.0K  5.0M   1% /run/lock
tmpfs           3.9G     0  3.9G   0% /sys/fs/cgroup
/dev/sda1        96M   29M   68M  30% /boot/efi
cgmfs           100K     0  100K   0% /run/cgmanager/fs
tmpfs           787M   72K  787M   1% /run/user/1000

How can i know which files takes the space?

**Comment by takluyver** [comment]
Wait, that's not even showing the same drive you showed originally. Is that on the same machine?

`du` can show you file size, and you can probably hack together either some Unix pipeline or a bit of Python code to sort it by size.

If you have GUI access to the machine, baobab is a tool that will show a neat radial chart of what is using space.

**Comment by dot27** [comment]
This is the relevant data.
I don't have gui access.

**Comment by dot27** [comment]
What do you recommend?

**Comment by Carreau** [comment]
`du -hs .` or  `du -hs *`  will give you space usage of all of your folders, including space used by subfolders. Then you can drill down to the bigger ones.

**Comment by ivanov** [comment]
Hey there, I'm going through old issues and it seems to me that it makes sense to close this one due to lack of activity. I think it is unlikely that IPython notebooks take up so much space on their own without some help from the user, so there isn't anything for us to do here.

Happy hacking! :bowtie:

**Comment by anthls** [comment]
I came across this thread from a Google search so others might too. Be aware that files deleted from within a Jupyter notebook (i.e. using menu functionality) are not deleted but moved to /home/<user>/.local/share/Trash/ (where <user> is the name of the user running the notebook, e.g. "ubuntu"). The files in the Trash directory can be permanently deleted using the "rm" command (and presumably other methods too). I removed the Trash files by using ssh to connect to the EC2 instance, cd to the Trash directory and executing rm.
